ci_framework/roles/reproducer/tasks/libvirt_layout.yml

```diff
@@ -0,0 +1,213 @@
+        dest: "/home/zuul/.kube/config"
+        content: >-
+          {{
+            (cifmw_use_devscripts | default(false) | bool) |
+            ternary(_devscripts_kubeconfig.content, _crc_kubeconfig.content) |
+            b64decode
+          }}
+        owner: zuul
+        group: zuul
+        mode: "0644"
+
+    - name: Inject kubeadmin-password if exists
+      when:
+        - _devscripts_kubeadm is defined
+      ansible.builtin.copy:
+        dest: "/home/zuul/.kube/kubeadmin-password"
+        content: "{{ _devscripts_kubeadm['content'] | b64decode }}"
+        owner: zuul
+        group: zuul
+        mode: "0600"
+
+    - name: Inject devscripts private key if set
+      when:
+        - _devscript_privkey is defined
+      ansible.builtin.copy:
+        dest: "/home/zuul/.ssh/devscripts_key"
+        content: "{{ _devscript_privkey['content'] | b64decode }}"
+        owner: "zuul"
+        group: "zuul"
+        mode: "0400"
+
+    - name: Inject CRC related content if needed
+      when:
+        - _layout.vms.crc is defined
+      block:
+        - name: Inject CRC ssh key
+          ansible.builtin.copy:
+            dest: "/home/zuul/.ssh/crc_key"
+            content: "{{ crc_priv_key['content'] | b64decode }}"
+            mode: "0400"
+            owner: zuul
+            group: zuul
+
+        - name: Inject crc related host entry
+          ansible.builtin.lineinfile:
+            path: /etc/hosts
+            line: >-
+              {{ cifmw_reproducer_crc_ip4 }} api.crc.testing
+              canary-openshift-ingress-canary.apps-crc.testing
+              console-openshift-console.apps-crc.testing
+              default-route-openshift-image-registry.apps-crc.testing
+              downloads-openshift-console.apps-crc.testing
+              oauth-openshift.apps-crc.testing
+
+    - name: RHEL repository setup for ansible-controller
+      when:
+        - ansible_facts['distribution'] == 'RedHat'
+      block:
+        - name: Get rhos-release
+          ansible.builtin.package:
+            name: "{{ cifmw_repo_setup_rhos_release_rpm }}"
+            state: present
+            disable_gpg_check: true
+
+        - name: Enable RHEL repos
+          ansible.builtin.command:
+            cmd: "rhos-release rhel"
+
+        - name: Install internal CA
+          ansible.builtin.package:
+            name: "{{ cifmw_reproducer_internal_ca }}"
+            disable_gpg_check: true
+
+    - name: Install some tools
+      ansible.builtin.package:
+        name:
+          - ansible-core
+          - bash-completion
+          - git-core
+          - make
+          - podman
+          - python3-jmespath
+          - python3-netaddr
+          - python3-pip
+          - tmux
+          - vim
+          - wget
+
+    - name: Install ansible dependencies
+      ansible.builtin.pip:
+        requirements: https://raw.githubusercontent.com/openstack-k8s-operators/ci-framework/main/ansible-requirements.txt
```

Comment:
Can we replace this with `make setup_molecule` and call it after push_code.yml get called?

Reply:
that command installs "a bit too much" imho for the use we have here.
With these couple of things (collections, dependencies), we're actually working in good intelligence with the base system. The `make setup_molecule` uses venv and a lot of other things we may actually not need outside of actual CI